storm-core/src/jvm/backtype/storm/networktopography/DefaultRackDNSToSwitchMapping.java

```diff
@@ -0,0 +1,39 @@
+package backtype.storm.networktopography;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This class implements the {@link DNSToSwitchMapping} interface
+ *    It returns the DEFAULT_RACK for every host.
+ */
+public final class DefaultRackDNSToSwitchMapping extends CachedDNSToSwitchMapping {
```

Comment:
I now we borrowed this from Hadoop, and it looks like Hadoop has the same issue, but we cannot subclass CachedDNSToSwitchMapping to get caching.  It does not work that way.  Please either fix CacheDNSToSwitchMapping to actually do caching when it is a parent class, or remove it all together and have it's children inherent directly from AbstractDNSToSwitchMapping.